content/en/tests/setup/java.md

```diff
@@ -348,12 +359,61 @@ public class ManualTest {
 
 Always call ``datadog.trace.api.civisibility.DDTestSession#end`` at the end so that all the test info is flushed to Datadog.
 
+## Best practices
+
+### Deterministic test parameters representation
+
+Test Visibility works best when the [test parameters are deterministic][8] and stay the same between test runs.
+If a test case has a parameter that varies between test executions (such as a current date, a random number, or an instance of a class whose `toString()` method is not overridden), some of the product features may not work as expected.
+For example, the history of executions may not be available, or the test case may not be classified as flaky even if it exhibits flakiness.
+
+The best way to fix this is to make sure that the test parameters are the same between test runs.
+
+In JUnit 5 this can also be addressed by [customizing the string representation of the test parameters][9] without changing their values.
```

Comment:
```suggestion
In JUnit 5, this can also be addressed by [customizing the string representation of the test parameters][9] without changing their values.
```